jvm/src/main/scala/io/glutenproject/expression/CoalesceOperatorTransformer.scala

```diff
@@ -38,7 +41,24 @@ class CoalesceTransformer(exps: Seq[Expression], original: Expression)
     with Logging {
 
   override def doTransform(args: java.lang.Object): ExpressionNode = {
-    throw new UnsupportedOperationException("Not supported: Coalesce.")
+    var notSupportFlag = false
+    val nodes = new java.util.ArrayList[ExpressionNode]()
+    val arrayBuffer = new ArrayBuffer[DataType]()
+    exps.foreach(expression => {
+      val expressionNode = expression.asInstanceOf[ExpressionTransformer].doTransform(args)
+      notSupportFlag = !expressionNode.isInstanceOf[ExpressionNode] || notSupportFlag
+      if (notSupportFlag) {
+        throw new UnsupportedOperationException(s"Not supported yet.")
+      }
+      arrayBuffer.append(expression.dataType)
+      nodes.add(expressionNode)
+    })
+    val functionMap = args.asInstanceOf[java.util.HashMap[String, java.lang.Long]]
+    val functionName = ConverterUtils.makeFuncName(ConverterUtils.COALESCE,
+      arrayBuffer, FunctionConfig.OPT)
+    val functionId = ExpressionBuilder.newScalarFunction(functionMap, functionName)
+    val typeNode = ConverterUtils.getTypeNode(exps.head.dataType, original.nullable)
+    ExpressionBuilder.makeScalarFunction(functionId, nodes, typeNode)
```

Comment:
Can you help rename this file into `CoalesceTransformer.scala` because it is not an operator.